src/main/java/com/xpn/xwiki/plugin/lucene/searcherProvider/SearcherProvider.java

```diff
@@ -38,82 +41,72 @@
    * List of Lucene indexes used for searching. By default there is only one such index
    * for all the wiki. One searches is created for each entry in {@link #indexDirs}.
    */
-  private Searcher[] searchers;
+  private final Searcher[] backedSearchers;
+
+  private volatile boolean markToClose;
 
-  private boolean markToClose;
+  private volatile boolean isClosed;
 
   private Set<Thread> connectedThreads;
```

Comment:
Why is this seperate `Set<> connectedThreads` required? Adding an entry directly to `ConcurrentMap<> connectedSearchResultsMap` upon connect() would be the same logically, easier to read and maintain.